File: Service/Admin/RMA.Service.Admin.MasterDataManager.Contracts/Entities/PensionGazetteResult.cs
using RMA.Service.Admin.MasterDataManager.Contracts.Enums;
using System;
using System.Collections.Generic;

namespace RMA.Service.Admin.MasterDataManager.Contracts.Entities
{
    public class PensionGazetteResult
    {
        public int PensionGazetteId { get; set; }
        public PensionGazetteTypeEnum PensionGazetteType { get; set; }
        public DateTime EffectiveFrom { get; set; }
        public DateTime EffectiveTo { get; set; }
        public List<PensionGazetteIncrease> Increases { get; set; }
    }

    public class PensionGazetteIncrease
    {
        public DateTime? IncidentMinDate { get; set; }
        public DateTime? IncidentMaxDate { get; set; }
        public PensionGazetteValueTypeEnum ValueType { get; set; }
        public decimal Value { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/shared-models-lib/src/lib/gazette/pension/pension-gazette-result.ts
import { PensionGazetteTypeEnum } from "../../enums/gazette/pension/pension-gazette-type-enum";
import { PensionGazetteValueTypeEnum } from "../../enums/gazette/pension/pension-gazette-value-type-enum";

export class PensionGazetteResult {
    pensionGazetteId: number;
    pensionGazetteType: PensionGazetteTypeEnum;
    effectiveFrom: Date;
    effectiveTo: Date;
    increases: PensionGazetteIncrease[];
}

export class PensionGazetteIncrease {
    incidentMinDate: Date;
    incidentMaxDate: Date;
    valueType: PensionGazetteValueTypeEnum;
    value: number;
}
